lib/p5.play.js

```diff
@@ -1480,72 +1422,80 @@ function Sprite(pInst, _x, _y, _w, _h) {
-      this.collider = new AABB(pInst, this.position, createVector(this.width, this.height));
-    } else if (type === 'rectangle' && arguments.length === 5) {
-      this.collider = new AABB(pInst, this.position, createVector(width, height), v);
-    } else if (type === 'circle' && arguments.length === 1) {
-      this.collider = new CircleCollider(pInst, this.position, Math.floor(Math.max(this.width, this.height) / 2));
-    } else if (type === 'circle' && arguments.length === 4) {
-      this.collider = new CircleCollider(pInst, this.position, width, v);
+    // Check correct arguments, provide context-sensitive usage message if wrong.
+    if (!(_type === 'point' || _type === 'circle' || _type === 'obb' || _type === 'aabb')) {
+      throw new TypeError('setCollider expects the first argument to be one of "point", "circle", "rectangle", "aabb" or "obb"');
+    } else if (_type === 'point' && !(arguments.length === 1 || arguments.length === 3)) {
+      throw new TypeError('Usage: setCollider("' + type + '"[, offsetX, offsetY])');
+    } else if (_type === 'circle' && !(arguments.length === 1 || arguments.length === 3 || arguments.length === 4)) {
+      throw new TypeError('Usage: setCollider("' + type + '"[, offsetX, offsetY[, radius]])');
+    } else if (_type === 'aabb' && !(arguments.length === 1 || arguments.length === 3 || arguments.length === 5)) {
+      throw new TypeError('Usage: setCollider("' + type + '"[, offsetX, offsetY[, width, height]])');
+    } else if (_type === 'obb' && !(arguments.length === 1 || arguments.length === 3 || arguments.length === 5 || arguments.length === 6)) {
+      throw new TypeError('Usage: setCollider("' + type + '"[, offsetX, offsetY[, width, height[, rotation]]])');
     }
 
-    quadTree.insert(this);
-  };
-
-  /**
-   * Returns a the bounding box of the current image
-   * @method getBoundingBox
-   */
-  this.getBoundingBox = function() {
-
-    var w = animations[currentAnimation].getWidth()*abs(this._getScaleX());
-    var h = animations[currentAnimation].getHeight()*abs(this._getScaleY());
-
-    //if the bounding box is 1x1 the image is not loaded
-    //potential issue with actual 1x1 images
-    if(w === 1 && h === 1) {
-      //not loaded yet
-      return new AABB(pInst, this.position, createVector(w, h));
-    }
-    else {
-      return new AABB(pInst, this.position, createVector(w, h));
+    //var center = this.position;
+    var offset = createVector(offsetX, offsetY);
+
+    if (_type === 'point') {
+      this.collider = p5.CollisionPoint.createFromSprite(this, offset);
+    } else if (_type === 'circle') {
+      this.collider = p5.CollisionCircle.createFromSprite(this, offset, width);
+    } else if (_type === 'aabb') {
+      this.collider = p5.AxisAlignedBoundingBox.createFromSprite(this, offset, width, height);
+    } else if (_type === 'obb') {
+      this.collider = p5.OrientedBoundingBox.createFromSprite(this, offset, width, height, radians(rotation));
```

Comment:
Naming nit: for the concrete colliders, I'd probably go with PointCollider, CircleCollider, AxisAlignedBoundingBoxCollider and OrientedBoundingBoxCollider, for consistency.